rust/crd/src/lib.rs

```diff
@@ -396,4 +574,173 @@ impl TrinoCluster {
             })?;
         Ok(format!("{}.0.0", product_version))
     }
+
+    /// Returns user provided authentication settings
+    pub fn get_authentication(&self) -> Option<&Authentication> {
+        let spec: &TrinoClusterSpec = &self.spec;
+        spec.authentication.as_ref()
+    }
+
+    /// Return user provided client TLS settings
+    pub fn get_client_tls(&self) -> Option<&TlsSecretClass> {
+        let spec: &TrinoClusterSpec = &self.spec;
+        spec.config.tls.as_ref()
+    }
+
+    /// Return if client TLS should be set depending on settings for authentication and client TLS.
+    pub fn tls_enabled(&self) -> bool {
+        self.get_authentication().is_some() || self.get_client_tls().is_some()
```

Comment:
```suggestion
        self.get_client_tls().is_some()
```

Since authn no longer implies TLS.